src/test/java/com/example/music_player/controller/SongControllerTest.java

```diff
@@ -0,0 +1,168 @@
+
+    @MockBean
+    private ISongService songService;
+
+    @MockBean
+    private ISourceService decorator;
+
+    @Autowired
+    private WebApplicationContext webApplicationContext;
+
+    @Test
+    void getAll() throws Exception {
+        when(songService.finedAllSongs()).thenReturn(Arrays.asList(song1, song2));
+        this.mockMvc.perform(get("/song/"))
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$", hasSize(2)))
+                .andExpect(jsonPath("$[*].id", containsInAnyOrder(1, 2)));
+    }
+
+    @Test
+    void findSongById() throws Exception {
+        when(songService.findSongById(anyLong()))
+                .thenReturn(new Song(1L, 1L, 1L, "name1", "notes1", 2001, "storageTypes1"));
+        this.mockMvc.perform(get("/song/getSong/1"))
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.id", equalTo(1)))
+                .andExpect(jsonPath("$.year", equalTo(2001)));
+    }
+
+    @Test
+    void addSong() throws Exception {
+        Mockito.when(songService.addSong(song1))
+                .thenReturn(song1.getId());
+        this.mockMvc.perform(post("/song/add/")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsString(song1)))
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andExpect(content().string(EXPECTED_CONTENT));
+    }
+
+    @Test
+    void updateAlbum() throws Exception {
+        when(songService.update(1L, song1)).thenReturn((song1.getId()));
+        this.mockMvc.perform(put("/song/update/1")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsString(song1)))
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andExpect(content().string(EXPECTED_CONTENT));
+    }
+
+    @Test
+    void deleteSong() throws Exception {
+        when(songService.deleteById(song1.getId())).thenReturn(String.valueOf(song1.getId()));
+        this.mockMvc.perform(delete("/song/deleteSong/1"))
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andExpect(content().string(EXPECTED_CONTENT));
+    }
+
+    @Test
+    void deleteByName() throws Exception {
+        when(songService.deleteSongByName(song1.getName())).thenReturn(song1.getName());
+        this.mockMvc.perform(delete("/song/deleteSongByName/name1"))
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andExpect(content().string("name1"));
+    }
+
+    @Test
+    void saveFile() throws Exception {
+        MockMultipartFile file = new MockMultipartFile(
+                "file",
+                "hello.txt",
+                MediaType.TEXT_PLAIN_VALUE,
+                "Hello, World!" .getBytes()
+        );
+        MockMvc mockMvc
```

Comment:
Is it required to create another mockMvc here?

Reply:
fixed it